modules/gdscript/gdscript_cache.cpp

```diff
@@ -338,9 +339,20 @@ Ref<GDScript> GDScriptCache::get_shallow_script(const String &p_path, Error &r_e
 }
 
 Ref<GDScript> GDScriptCache::get_full_script(const String &p_path, Error &r_error, const String &p_owner, bool p_update_from_disk) {
+	Ref<GDScript> uncached_script = get_full_script_no_resource_cache(p_path, r_error, p_owner, p_update_from_disk);
+
+	// Resources don't know whether they are cached, so using set_path after set_path_cache does not add the resource to the cache if the path is the same.
+	// We reset the cached path from get_shallow_script so that when ResourceLoaders uses set_path everything can get cached correctly.
```

Comment:
```suggestion
	// Resources don't know whether they are cached, so using `set_path()` after `set_path_cache()` does not add the resource to the cache if the path is the same.
	// We reset the cached path from `get_shallow_script()` so that when `ResourceLoaders` uses `set_path()` everything can get cached correctly.
```